h2o-algos/src/test/java/hex/tree/gbm/GBMRowToTreeAssignmentTest.java

```diff
@@ -0,0 +1,268 @@
+package hex.tree.gbm;
+
+import hex.genmodel.utils.DistributionFamily;
+import hex.tree.CompressedTree;
+import hex.tree.Sample;
+import org.apache.log4j.Logger;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.EnvironmentVariables;
+import org.junit.runner.RunWith;
+import water.DKV;
+import water.Key;
+import water.Scope;
+import water.TestUtil;
+import water.fvec.Frame;
+import water.fvec.TestFrameBuilder;
+import water.fvec.Vec;
+import water.runner.CloudSize;
+import water.runner.H2ORunner;
+
+import java.io.File;
+
+import static org.junit.Assert.assertNotNull;
+
+
+@CloudSize(1)
+@RunWith(H2ORunner.class)
+public class GBMRowToTreeAssignmentTest extends TestUtil {
+    private static final Logger LOG = Logger.getLogger(GBMRowToTreeAssignmentTest.class);
+
+    @Rule
+    public EnvironmentVariables environmentVariables = new EnvironmentVariables();
+
+    @Before
+    public void setup() {
+        final File h2oHomeDir = new File(System.getProperty("user.dir")).getParentFile();
+        environmentVariables.set("H2O_FILES_SEARCH_PATH", h2oHomeDir.getAbsolutePath());
+    }
+
+    @Test
+    public void testRowToTreeAssignmentSmokeBinomialHardcoded() {
+        Scope.enter();
+        try {
+            Scope.enter();
```

Comment:
Do we need to enter the scope multiple times?

Reply:
No, fixed 👍